package tests;

import org.testng.Assert;
import org.testng.annotations.Test;

import base.BaseTest;
import base.DriverFactory;
import pages.LoginPage;
import utils.ConfigReaderUtils;
import utils.ReportUtil;
import utils.ScreenshotUtil;

public class LoginTest extends BaseTest {

	
    @Test
    public void loginTest() throws InterruptedException {
        ReportUtil.logInfo("üîê Starting login test");

        try {
            new LoginPage(DriverFactory.getDriver()).login(ConfigReaderUtils.get("username"), ConfigReaderUtils.get("password"));
            ReportUtil.logInfo("üîç Verifying dashboard access");

            Assert.assertTrue(DriverFactory.getDriver().getCurrentUrl().contains("dashboard"), "‚ùå Dashboard not loaded");
            

            String screenshotPath = ScreenshotUtil.capture(DriverFactory.getDriver(), "login_success");
            ReportUtil.logPass("‚úÖ Login successful", screenshotPath);

        } catch (AssertionError ae) {
            String screenshotPath = ScreenshotUtil.capture(DriverFactory.getDriver(), "login_assert_fail");
            ReportUtil.logFail("‚ùå Assertion failed: " + ae.getMessage(), screenshotPath);
            throw ae;

        } catch (Exception e) {
            String screenshotPath = ScreenshotUtil.capture(DriverFactory.getDriver(), "login_exception");
            ReportUtil.logFail("‚ö†Ô∏è Unexpected error: " + e.getMessage(), screenshotPath);
            throw new RuntimeException(e);
        }
    }
}